### ISSUE TITLE
Always display exit numbers

### ISSUE THREAD

**Issue by jm355** [osmandapp/OsmAnd#20685]
### Describe the idea

Currently the map displays the exit number on the highway about 1.6 miles away from the exit, I think it should always be desplayed when it's the next turn

### Expected behaviour

When the next turn is to take a numbered exit, the exit number should always be visible

### Alternatives you've considered

Organic maps and magic earth does this

### Context

Screenshots of current behavior

No exit number:
![Screenshot_20240825-125245](https://github.com/user-attachments/assets/fc9cc36c-5c45-4b95-85f7-8fe8548f028e)

exit number (blue square, exit number 1 in the top left):
![Screenshot_20240825-125252](https://github.com/user-attachments/assets/34a2efd5-f547-4abd-9a7d-b9378274fd3b)

**Comment by vshcherb** [comment]
We plan to separate it into 2 widgets current street name and next street name, in that case exit will be always displayed

**Comment by Max1234-Ita** [comment]
> ### Describe the idea
> 
> Currently the map displays the exit number on the highway about 1.6 miles away from the exit, I think it should always be desplayed when it's the next turn

Also make sure the exit number is actually present in the map, i.e. check with an OSM editor such as iD or JOSM.
An information missing in the map data cannot be displayed...

**Comment by jm355** [comment]
> > ### Describe the idea
> > Currently the map displays the exit number on the highway about 1.6 miles away from the exit, I think it should always be desplayed when it's the next turn
> 
> Also make sure the exit number is actually present in the map, i.e. check with an OSM editor such as iD or JOSM. An information missing in the map data cannot be displayed...

Yep, if the map data is missing then it can't display data it doesn't have. In this example, you can see in the second screenshot the data exists and it does get displayed eventually, but isn't always visible for the duration of the time that the exit is the next turn

**Comment by jm355** [comment]
@vshcherb are you talking about the new Route guidance->Next turn top/bottom widget? At least on 4.9.3, that widget doesn't seem to show different information to the "Street name" widget. It still waits until you're close to an exit to display the exit number and upcoming street name. 
Also, the big font and bright colors of the old "Street name" widget are a lot easier to see for me
![Screenshot_20241101-115620](https://github.com/user-attachments/assets/99eeddf0-adf6-4190-b14e-e44808ac3eec) ![Screenshot_20241101-114730](https://github.com/user-attachments/assets/9478cff4-e35c-4b58-b893-2a65ac9ca8b4)

**Comment by jm355** [comment]
Ah gotcha, cool! 

I noticed today, it was showing the exit numbers of the nearest exit and saying to go straight/**not** take the exit, I think that's really confusing. I want to be able to look down, see an exit number and know that's the exit I need to get off on. 

The way it is now, I look down, see an exit number, and then have to look around on the screen for context clues to tell me if I'm taking that exit or not. 

Not to mention, that's redundant info: if I know which exit I should take, I also know every exit I shouldn't take. But if I know one exit I shouldn't take, it tells me nothing about any other exit
